### PR TITLE
split sync test and limit mining time in the tests

### PR DESCRIPTION
**Pull request by winsvega**
mining limit for tests

### CODE REVIEW

File: test/boostTest.cpp [hunk 1]
```diff
@@ -83,6 +83,17 @@ int main( int argc, char* argv[] )
 {
 	//Initialize options
 	dev::test::Options const& opt = dev::test::Options::get(argc, argv);
+
+	char* path = (argv[0]);
+	argc = 3;
```
Comment (chriseth) [inline test/boostTest.cpp]: We should not just ignore the options that are passed here. Also I think it is really bad style to modify `argv`.

File: test/libethereum/BlockChain.cpp [hunk 2]
```diff
@@ -60,17 +61,9 @@ BOOST_FIXTURE_TEST_SUITE(BlockChainSuite, TestOutputHelper)
 BOOST_AUTO_TEST_CASE(opendb)
 {
 	TestBlock genesis = TestBlockChain::defaultGenesisBlock();
-
 	TransientDirectory tempDirBlockchain;
 	ChainParams p(genesisInfo(Network::Test), genesis.bytes(), genesis.accountMap());
 	BlockChain bc(p, tempDirBlockchain.path(), WithExisting::Kill);
-
-	Block block = bc.genesisBlock(genesis.state().db());
```
Comment (chriseth) [inline test/libethereum/BlockChain.cpp:68]: Does this still test enough if you remove these lines?

File: test/boostTest.cpp (hunk 1 above)
Comment (winsvega) [inline test/boostTest.cpp]: the only way I found it work without issues. It is just to test on travis only this test suite. 
will not be included to the final PR

File: test/libethereum/BlockChain.cpp [hunk 3]
```diff
@@ -108,106 +101,210 @@ BOOST_AUTO_TEST_CASE(opendb)
...
+		{
+			BOOST_ERROR("Exception thrown when trying to mine or import a block!");
+		}
+	};
+
+	testMiningFunc(threadFunc);
 }
+
+// Temporary disable of "sync" test which is repeatedly failing in TravisCI for Ubuntu Trusty.
+#if !defined(ETH_AFTER_REPOSITORY_MERGE)
```
Comment (chriseth) [inline test/libethereum/BlockChain.cpp]: Please remove this.

**Comment by chriseth** [comment]
If this works, it might be a good workaround!

File: test/libethereum/BlockChain.cpp (hunk 2 above)
Comment (winsvega) [inline test/libethereum/BlockChain.cpp:68]: Yes. Just open same DB twice but without mining.

**Comment by chriseth** [comment]
Still failing. I would like to see at least one successful run before we can merge it. Re-triggered travis.

**Comment by winsvega** [comment]
bob has changed travis wait option to 40 minutes. it should work now

**Comment by winsvega** [comment]
ok so I think it could be merged without last 2 commits (merge and remove)

**Comment by bobsummerwill** [comment]
Make it so, @winsvega !

**Comment by winsvega** [comment]
actually I already thought so 9 days ago )

**Pull request merged**